comp/process/apiserver/apiserver_test.go

```diff
@@ -0,0 +1,32 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package apiserver
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/fx"
+
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+)
+
+func TestLifecycle(t *testing.T) {
```

Comment:
Is there a way to assess that the API Server component has been correctly shut down?

Reply:
So - currently the answer is yes and no. `fxutil.Test` does require that the the components that are started successfully call their `OnStop` hook once the test finishes ([code](https://github.com/DataDog/datadog-agent/blob/73c8bfe45834b674cdffdc11003e854ab861ffed/pkg/util/fxutil/test.go#L42-L44)). However, currently our `OnStop` hook is a noop, as seen by @ogaca-dd's comment above https://github.com/DataDog/datadog-agent/pull/16343#discussion_r1159685791. I did this because a `StopApiServer` method actually never existed in the first place. From your folks feedback, I think it might be the perfect time to add one though 😄